aten/src/ATen/native/EmbeddingBag.cpp

```diff
@@ -1271,11 +1272,11 @@ Tensor _embedding_bag_backward(const Tensor &grad, const Tensor &indices_,
 
   Tensor offset2bag_;
   if (indices.numel() != 0 && offset2bag.numel() == 0) {
-    offset2bag_ = at::zeros(
-       {indices.size(0) + 1}, offsets.options()); // offset2bag = [0 0 0 0 0]
+    offset2bag_ = offsets.new_zeros(
+      {indices.size(0) + 1}, offsets.options()); // offset2bag = [0 0 0 0 0]
 
     make_offset2bag(offsets, offset2bag_);
-    offset2bag_.resize_({indices.size(0)});
+    offset2bag_ = offset2bag_.narrow(0, 0, indices.size(0));
```

Comment:
Sorry, on second thought:
- it is faster to do the resize_() because that does not involve creating a new Tensor
- because it is faster, we should try to not to regress performance here.
- so we should still have a isTensorSubclassLike check for this line (to switch between narrow and resize_.

This doesn't apply to `zeros` vs `new_zeros` because those take roughly the same amount of time.

What do you think?

Reply:
I had a question, if I am correct, `narrow` will return view, right? That should be fast then, right?

Thanks!

Reply:
Narrow will return a view. A view is a brand-new Tensor but with the same storage as before. The allocation of the brand-new Tensor could be expensive (but not as expensive as copying all of the data) -- around 1-3us.

Reply:
Updated. Thanks :)